File: Example/common/src/main/java/com/example/common/okhttp/download/FileManager.java
package com.example.common.okhttp.download;

import android.annotation.SuppressLint;
import android.content.Context;

import java.io.File;

/**
 * @author tengfei
 * date 2019/2/16 9:31 PM
 * email tengfeigo@outlook.com
 * description
 */
public class FileManager {

    private File mRootFile;
    private Context mContext;

    private FileManager(){}

    public static FileManager getInstance(){
        return Holder.FILE_MANAGER;
    }

    private static class Holder{
        @SuppressLint("StaticFieldLeak")
        private static final FileManager FILE_MANAGER = new FileManager();
    }

    public void init(Context context) {
        this.mContext = context.getApplicationContext();
    }

    public void rootDir(File file) {
        if (!file.exists()) {
            //noinspection ResultOfMethodCallIgnored
            file.mkdirs();
        }
        if (file.exists() && file.isDirectory()) {
            mRootFile = file;
        }
    }

    public File getFile(String url) {
        String fileName = Utils.md5Uri(url);
        if (mRootFile == null) {
            mRootFile = mContext.getCacheDir();
        }
        return new File(mRootFile, fileName);
    }
}

File: OkHttpExample/app/src/main/java/com/example/okhttp/FileManager.kt
package com.example.okhttp

import android.annotation.SuppressLint
import android.content.Context
import java.io.File

/**
 * @author tengfei
 * date 2019/3/6 10:14 PM
 * email tengfeigo@outlook.com
 * description File管理类
 */
class FileManager private constructor() {

    var context: Context? = null
    private var mRootDir: File? = null

    companion object {
        @SuppressLint("StaticFieldLeak")
        val instance = Holder.fileManager
    }

    private object Holder {
        @SuppressLint("StaticFieldLeak")
        val fileManager = FileManager()
    }

    fun initFileManager(context: Context) {
        this.context = context.applicationContext
    }

    fun rootDir(file: File) {
        if (!file.exists()) {
            file.mkdirs()
        }

        if (file.exists() && file.isDirectory) {
            mRootDir = file
        }
    }

    fun createFile(): File {
        val fileName = ""
        if (mRootDir == null) {
            mRootDir = context!!.cacheDir
        }

        return File(mRootDir, fileName)
    }
}